resources/idlharness.js

```diff
@@ -218,6 +226,81 @@ IdlArray.prototype.add_untested_idls = function(raw_idls, options)
     this.internal_add_idls(parsed_idls, options);
 };
 
+//@}
+IdlArray.prototype.add_dependency_idls = function(raw_idls, options)
+//@{
+{
+    const parsed_idls = WebIDL2.parse(raw_idls);
+    const new_options = { only: [] }
+    
+    const all_deps = new Set();
+    Object.values(this["inheritance"]).forEach(v => all_deps.add(v));
+    Object.values(this["implements"]).forEach(v => all_deps.add(v));
+    this.partials.map(p => p.name).forEach(v => all_deps.add(v));
+    
+    console.log('all deps', all_deps);
+
+    if (options && options.except && options.only) {
+        throw new IdlHarnessError("The only and except options can't be used together.");
+    }
+    const skip = name => {
+        return !name
+            || options && (
+                options.except && options.except.includes(name)
+                || options.only && !options.only.includes(name));
+    }
+
+    // Record of skipped items in case we later determine they are a dependency.
+    const skipped = new Map();
+    const process = function(parsed) {
+        let name = parsed.name
+            || parsed.type == "implements" && parsed.target
+            || parsed.type == "includes" && parsed.target;
+        if (skip(name) || !all_deps.has(name)) {
+            name && skipped.set(name, parsed);
+            return;
+        }
+
+        console.log(parsed);
+        
+        new_options.only.push(name);
+        const follow_up = [];
+        for (const dep_type of ["inheritance", "implements", "includes"]) {
+            if (parsed[dep_type]) {
+                const dep = parsed[dep_type];
+                new_options.only.push(dep);
+                all_deps.add(dep);
+                follow_up.push(dep);
+                console.log(`Added ${dep_type} dep ${dep}`);
+            }
+        }
+
+        if (follow_up && skipped.has(follow_up)) {
```

Comment:
This seems broken. `follow_up` is an array, and even an empty array is truthy in JS (unlike in Python). Should have been a loop?